src/js/components/AppPageComponent.jsx

```diff
@@ -273,15 +282,148 @@ var AppPageComponent = React.createClass({
     );
   },
 
+  getVolumeStatus: function () {
+    var volume = this.getVolume();
+
+    if (volume == null) {
+      return null;
+    }
+
+    var className = classNames("volume-status", {
+      "volume-attached": volume.status === "Attached"
+    });
+
+    return (
+      <h2 className={className}>{volume.status}</h2>
+    );
+  },
+
+  getVolume: function () {
+    var {app, volumeId} = this.state;
+
+    if (app.tasks != null) {
+      return null;
+    }
+
+    var tasks = app.tasks;
+    return tasks
+      // Get the first volume from a task with the same id as provided
+      // by the router. This should be unique.
+      .reduce((memo, task) => {
+        if (memo == null) {
+          return task.localVolumes
+            .filter(volume => volume.persistenceId === volumeId)
+            .reduce((memo, volume) => {
+              if (memo != null) {
+                return memo;
+
+              }
+              volume.taskId = task.id;
+              volume.host = task.host;
+              volume.status = task.status == null
+                ? "Detached"
+                : "Attached";
+              return volume;
+            }, null);
+        }
+        return memo;
+      }, null);
+  },
+
+  getVolumeDetails: function () {
+    var {app, appId} = this.state;
+
+    var volume = this.getVolume();
+
+    if (volume == null) {
+      return null;
+    }
+
+    if (app.container != null && app.container.volumes != null) {
+      app.container.volumes.forEach(function (volumeConfig) {
+        if (volumeConfig.containerPath &&
+            volumeConfig.containerPath === volume.containerPath) {
+          Object.keys(volumeConfig).forEach(key =>
+            volume[key] = volumeConfig[key]
+          );
+        }
+      });
+    }
+
+    var taskURI = "#apps/" +
+      encodeURIComponent(this.state.appId) +
+      "/" + encodeURIComponent(volume.taskId);
+
+    var appURI = "#apps/" +
+      encodeURIComponent(this.state.appId);
+
+    return (
+      <dl className={"dl-horizontal"}>
+        <dt>Container Path</dt>
+        <dd>{volume.containerPath}</dd>
+        <dt>Mode</dt>
+        <dd>{volume.mode}</dd>
+        <dt>Size (MiB)</dt>
+        <dd>{volume.persistent.size}</dd>
+        <dt>Application</dt>
+        <dd>
+          <a href={appURI}>{appId}</a>
+        </dd>
+        <dt>Task Id</dt>
+        <dd>
+          <a href={taskURI}>{volume.taskId}</a>
+        </dd>
+        <dt>Host</dt>
+        <dd>
+          {volume.host}
+        </dd>
+      </dl>
+    );
+  },
+
   getAppDetails: function () {
     var state = this.state;
     var model = state.app;
 
+    var volumes = model.tasks.reduce((memo, task) => {
```

Comment:
Is it possible to move this out of the AppPageComponent-file?
Looks like something a store should do?